src/azul/collections.py

```diff
@@ -271,3 +273,64 @@ def to_dict(self):
             k: v.to_dict() if isinstance(v, NestedDict) else v
             for k, v in self.items()
         }
+
+
+class OrderedSet(MutableSet[Any]):
+    """
+    A mutable set that maintains insertion order. It is not a sequence.
+
+    >>> OrderedSet() == set()
+    True
+    >>> 'a' in OrderedSet(['a', 'b'])
+    True
+    >>> OrderedSet(['a', 'b', 'c', 'b'])
+    OrderedSet(['a', 'b', 'c'])
+    """
+
+    def __init__(self, members: Iterable = (), /) -> None:
+        self.inner: dict = dict.fromkeys(members)
+
+    def __repr__(self) -> str:
+        if not self:
+            return f"{self.__class__.__name__}()"
+        return f"{self.__class__.__name__}({list(self)!r})"
+
+    def __iter__(self) -> Iterator:
+        return iter(self.inner.keys())
+
+    def __len__(self) -> int:
+        return len(self.inner)
+
+    def __eq__(self, member: object) -> bool:
```

Comment:
`member` is a deceptive name for the argument here since we're not checking for membership.
```suggestion
    def __eq__(self, other: K) -> bool:
```

Reply:
Note that I changed the argument name to the suggested, but not the type hint. I think the comparison method may take as input other class types e.g `OrderedSet() == set()`, let me know if you disagree.

Reply:
I had meant to write `AbstractSet[K]` instead of just `K`, but it [looks like](https://stackoverflow.com/a/37557540/1530508) this is not correct, so you're good :+1: